### PR TITLE
[Expo] Add v 27.0.0 docs

### PR DESCRIPTION
**Pull request by moshfeu**
Please fill in this template.

- [X] Use a meaningful title for the pull request. Include the name of the package modified.
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [X] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [X] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://docs.expo.io/versions/v27.0.0/>>
- [ ] Increase the version number in the header if appropriate.
- [X] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by armanio123** [comment]
Seems like the typescript-bot got confused and didn't mentioned the type owners for review. @KonstantinKai @martynaskadisa @janaagaard75 @ssanchezmarc @fhelwanger @umidbekkarimov Could any of you please do a review and let me know if it's good for merge?

File: types/expo/index.d.ts [hunk 1]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+ * https://docs.expo.io/versions/latest/sdk/haptic
+ * Provides haptic feedback for iOS 10+ devices using the Taptic Engine.
+ * If this is used in Android the device will use ReactNative.Vibrate instead, it's best to just avoid this.
+ */
+export namespace Haptic {
+  /**
+   * Used to let a user know when a selection change has been registered
+   */
+  function selection(): void;
+  function notification(type: NotificationFeedbackType): void;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: `type` argument is optional -> https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/Haptic.js#L68

File: types/expo/index.d.ts [hunk 2]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+  function impact(style: ImpactFeedbackStyle): void;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: `style` argument is also optional -> https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/Haptic.js#L80

File: types/expo/index.d.ts [hunk 3]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+
+  enum ImpactFeedbackStyle {
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/Haptic.js#L40
I think, preferably, need to export enum with `ImpactStyles` name, instead `ImpactFeedbackStyle`. And use this enum like arg type in the method above.

File: types/expo/index.d.ts [hunk 4]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    Light = 'light',
+    Medium = 'medium',
+    Heavy = 'heavy'
+  }
+
+  enum NotificationFeedbackType {
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/Haptic.js#L29
Similar to the previous comment.

File: types/expo/index.d.ts [hunk 5]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    duration = 'duration'
+  }
+
+  // region Asset
+  interface AssetAndroid {
+    albumId: string;
+  }
+
+  interface AssetIos {
+    mediaSubtypes: MediaType[];
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop.

File: types/expo/index.d.ts [hunk 6]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    modificationTime = 'modificationTime',
+    mediaType = 'mediaType',
+    width = 'width',
+    height = 'height',
+    duration = 'duration'
+  }
+
+  // region Asset
+  interface AssetAndroid {
+    albumId: string;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

File: types/expo/index.d.ts [hunk 7]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    filename: string;
+    uri: string;
+    mediaType: string;
+    width: number;
+    height: number;
+    creationTime: number;
+    modificationTime: number;
+    duration: number;
+    // *
+    localUri: string;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

File: types/expo/index.d.ts [hunk 8]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    // *
+    location: object;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop with type Location.LocationProps

File: types/expo/index.d.ts [hunk 9]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    // *
+    exif: object;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

File: types/expo/index.d.ts [hunk 10]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+  }
+
+  /**
+   * These fields can be obtained only by calling getAssetInfoAsync method
+   */
+  //#endregion
+
+  // #region Album
+  interface AlbumIos {
+    type: string;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

File: types/expo/index.d.ts [hunk 11]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+    // *
+    startTime: number;
+    // *
+    endTime: number;
+    // *
+    approximateLocation: Location.LocationProps;
+    // *
+    locationNames: string[];
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

File: types/expo/index.d.ts [hunk 12]
```diff
@@ -2852,3 +2859,189 @@ export namespace Updates {
...
+
+  // #region Album
+  interface AlbumIos {
+    type: string;
+    // *
+    startTime: number;
+    // *
+    endTime: number;
+    // *
+    approximateLocation: Location.LocationProps;
```
Comment (KonstantinKai) [inline types/expo/index.d.ts]: Optional prop

**Review by KonstantinKai — CHANGES_REQUESTED** [review]
@moshfeu, thanks a lot for contribution. But, I found a few inconsistencies with the original source.
https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/MediaLibrary.js
https://github.com/expo/expo-sdk/blob/191eeffb934c84ffb7e05dd1c5616282dccaff23/src/Haptic.js

**Review by KonstantinKai — CHANGES_REQUESTED** [review]
@moshfeu, please make changes according to the source.

**Comment by moshfeu** [comment]
@KonstantinKai I pushed a fix commit for all of your good comments. Can you take a look? 
Thanks!

**Review by KonstantinKai — APPROVED** [review]
@moshfeu looks good. Approved. 👍

**Comment by moshfeu** [comment]
@KonstantinKai Thanks!

**Pull request merged**